src/containers/catalog/withCatalogItems.js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Query } from "react-apollo";
+import { Provider } from "mobx-react";
+import primaryShopIdQuery from "../common-gql/primaryShopId.gql";
+import catalogItemsQuery from "./catalogItems.gql";
+
+/**
+ * withCatalogItems higher order query component for fetching primaryShopId and catalog data
+ * @name withCatalogItems
+ * @param {React.Component} Component to decorate and apply
+ * @returns {React.Component} - component decorated with primaryShopId and catalog as props
+ */
+export default (Component) => (
+  class CatalogItems extends React.Component {
+    render() {
+      return (
+        <Query query={primaryShopIdQuery}>
```

Comment:
The primaryShopId is already available in the context, so we can avoid making an additional query to the endpoint.

The following or something similar
```
@inject("primaryShopId")
@observer
class CatalogItems
```

then `this.props.primaryShopId`